sdk/core/Azure.Core/src/Serialization/AzureCoreExtensions.cs

```diff
@@ -44,17 +44,17 @@ public static class AzureCoreExtensions
             (T?)await serializer.DeserializeAsync(data.ToStream(), typeof(T), cancellationToken).ConfigureAwait(false);
 
         /// <summary>
-        /// Converts the <see cref="BinaryData"/> to a Dictionary of string to object.
-        /// Each value in the key value pair will be strongly typed as an int, long, string, Guid, double or bool.
-        /// Each value can also be another Dictionary of string object representing an inner object or
-        /// a List of objects representing an array.
+        /// Converts the <see cref="BinaryData"/> to an object based on the json value it contains.
+        /// If it is a primitive type such as string, int, double, bool, or null literal you will get that back.
+        /// Otherwise you will get back either an array of object or Dictionary of string to object.
+        /// Each value in the key value pair or list will be also be converted into a primitive or another complex type recursively.
         /// </summary>
         /// <param name="data">The <see cref="BinaryData"/> instance to convert.</param>
         /// <returns>The data converted to the Dictionary of string to object.</returns>
```

Comment:
```suggestion
        /// Converts the json value represented by <see cref="BinaryData"/> to an object of a specific type.
        /// If it is a primitive type such as string, int, double, bool, or null literal you will get that back.
        /// Otherwise you will get back either an array of object or Dictionary of string to object.
        /// Each value in the key value pair or list will be also be converted into a primitive or another complex type recursively.
        /// </summary>
        /// <param name="data">The <see cref="BinaryData"/> instance to convert.</param>
       /// <returns> The object value of the json value.
        /// If the object contains a primitive type such as string, int, double, bool, or null literal, it returns that type.
        /// Otherwise, it returns either an <see cref="object[]"/> or <see cref="Dictionary{TKey, TValue}"/> where TKey is <see cref="string"/> and TValue is <see cref="object"/>.
        /// Each value in the key value pair or list will be also be converted into a primitive or another complex type.
        /// </returns>
```